app/controllers/pokemons_controller.rb

```diff
@@ -21,7 +21,8 @@ def edit
 
   # POST /pokemons or /pokemons.json
   def create
-    @pokemon = Pokemon.new(pokemon_params)
+    #Call the function to setup the image appending
+    @pokemon = image_process()
```

Comment:
This is a nice try, but we should keep the `@pokemon = Pokemon.new(pokemon_params)` in the create method. The piece that should be pulled out is the image attach. This follows the idea that "each method should do one thing". So instead of create grabbing a pokemon, attaching an image to it, then saving it, it will just grab a pokemon and save it and we can delegate the image attaching part to another method.

Reply:
Okay, I rewrote that function and hope this follow the methodology of "each method should do one thing".